tool/plugin/lib/plugin.dart

```diff
@@ -532,73 +532,40 @@ class BuildCommand extends ProductCommand {
         }
       }
 
-      // TODO(devoncarew): Remove this when we no longer support AS 3.1.
-      File processedFile1, processedFile2, processedFile3;
-      String oldSource1, oldSource2, oldSource3, newSource;
-      if (spec.version == '2017.3') {
-        processedFile1 = new File(
-            'flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java');
-        log('Editing ${processedFile1.path}');
-        oldSource1 = processedFile1.readAsStringSync();
-        newSource = oldSource1;
-        newSource = newSource.replaceAll(
-          'import com.android.tools.idea.npw.model.NewModuleModel',
-          'import com.android.tools.idea.npw.module.NewModuleModel',
-        );
-        newSource = newSource.replaceAll(
-          'import java.awt.*',
-          'import javax.swing.*',
-        );
-        newSource = newSource.replaceAll('public Image', 'public Icon');
-        newSource =
-            newSource.replaceAll('IconUtil.toImage', ''); // Leaves parens
-        processedFile1.writeAsStringSync(newSource);
-        processedFile2 = new File(
-            'flutter-studio/src/io/flutter/project/ChoseProjectTypeStep.java');
-        log('Editing ${processedFile2.path}');
-        oldSource2 = processedFile2.readAsStringSync();
-        newSource = oldSource2;
-        newSource = newSource.replaceAll(
-            "import com.intellij.ui.components.JBScrollPane;",
-            "import com.intellij.ui.components.JBScrollPane;\nimport com.intellij.util.IconUtil;");
-        newSource = newSource.replaceAll(
-            'image.getIcon()', 'IconUtil.toImage(image.getIcon())');
-        processedFile2.writeAsStringSync(newSource);
-        processedFile3 = new File(
-            'flutter-studio/src/io/flutter/project/FlutterProjectSystem.java');
-        oldSource3 = processedFile3.readAsStringSync();
-        newSource = oldSource3;
-        newSource = newSource.replaceAll('.LightResourceClassService', '.*');
-        newSource =
-            newSource.replaceAll(' LightResourceClassService', ' Object');
-        processedFile3.writeAsStringSync(newSource);
-      } else if (spec.version == '2018.1') {
+      // TODO: Remove this when we no longer support AS 3.2.
+      File processedFile3, processedFile4;
+      String oldSource3, oldSource4, newSource;
```

Comment:
I know you didn't add it but it could be more robust to have
Map<String, File> oldSources;
so you can ensure you don't forget to restore an old source file or have a similar typo.

Reply:
Good point, and it was really hard to debug when that happened. Was hoping the problem would go away but I no longer think it will. We might want to see if this can be made data-driven from entries in the product matrix.